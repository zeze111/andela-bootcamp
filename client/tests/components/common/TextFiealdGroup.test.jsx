import React from 'react';

import {
  TextFieldGroup,
  TextFieldGroup2,
  TextFieldGroup3,
  LinkFieldGroup,
  LinkFieldGroup2
} from '../../../components/common/TextFieldGroup';

const props = {
  value: '',
  onChange: jest.fn(),
  id: '',
  type: '',
  name: '',
  label: '',
  error: [],
  icon: '',
  to: '',
  href: '',
  dataTool: '',
  item: '',
}

describe('Test for TextFieldGroup', () => {
  it('should render component', () => {
    const wrapper = shallow(<TextFieldGroup {...props}/> );
    expect(wrapper).toMatchSnapshot()
  });
});

describe('Test for TextFieldGroup2', () => {
  it('should render component', () => {
    const wrapper = shallow(<TextFieldGroup2 {...props}/> );
    expect(wrapper).toMatchSnapshot()
  });
});

describe('Test for TextFieldGroup', () => {
  it('should render component', () => {
    const wrapper = shallow(<TextFieldGroup3 {...props}/> );
    expect(wrapper).toMatchSnapshot()
  });
});

describe('Test for LinkFieldGroup', () => {
  it('should render component', () => {
    const wrapper = shallow(<LinkFieldGroup {...props}/> );
    expect(wrapper).toMatchSnapshot()
  });
});

describe('Test for LinkFieldGroup2', () => {
  it('should render component', () => {
    const wrapper = shallow(<LinkFieldGroup2 {...props}/> );
    expect(wrapper).toMatchSnapshot()
  });
});
