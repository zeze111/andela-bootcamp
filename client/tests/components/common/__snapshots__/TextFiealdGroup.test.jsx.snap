// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for LinkFieldGroup should render component 1`] = `
<li>
  <Link
    className="btn-floating btn-medium tooltipped waves-effect waves-light grey"
    data-position="bottom"
    data-tooltip=""
    href=""
    replace={false}
    to=""
  >
    <i
      className="material-icons"
    />
  </Link>
</li>
`;

exports[`Test for LinkFieldGroup2 should render component 1`] = `
<li
  className="nav-list"
>
  <i
    className="material-icons"
  />
  <Link
    className="white-text list-item caps"
    href=""
    replace={false}
    to=""
  />
</li>
`;

exports[`Test for TextFieldGroup should render component 1`] = `
<div
  className="input-field"
>
  <label
    className=""
    htmlFor=""
  >
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       
       
    </span>
  </label>
  <input
    className="validate"
    id=""
    name=""
    onChange={[MockFunction]}
    placeholder=""
    type=""
    value=""
  />
  <span
    className="help-block red-text error-text-field"
  />
</div>
`;

exports[`Test for TextFieldGroup should render component 2`] = `
<div
  className="input-field"
>
  <label
    className="active"
    htmlFor=""
  >
    <i
      className="material-icons left"
    />
  </label>
  <input
    className="validate"
    disabled={false}
    id=""
    name=""
    onChange={[MockFunction]}
    type=""
    value=""
  />
  <span
    className="help-block red-text error-text-field"
  />
</div>
`;

exports[`Test for TextFieldGroup2 should render component 1`] = `
<div
  className="input-field"
>
  <label
    className="active"
    htmlFor=""
  >
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       
       
    </span>
  </label>
  <textarea
    className="materialize-textarea"
    id=""
    name=""
    onChange={[MockFunction]}
    placeholder=""
    type="text"
    value=""
  />
  <span
    className="help-block red-text error-text-field"
  />
</div>
`;
